grype/matcher/apk/matcher.go

```diff
@@ -23,39 +26,109 @@ func (m *Matcher) Type() match.MatcherType {
 func (m *Matcher) Match(store vulnerability.Provider, d *distro.Distro, p pkg.Package) ([]match.Match, error) {
 	var matches = make([]match.Match, 0)
 
-	// map {  CVE string : []match }
-	var secDbCandidates = make(map[string][]match.Match)
-
 	// find Alpine SecDB matches for the given package name and version
 	secDbMatches, err := common.FindMatchesByPackageDistro(store, d, p, m.Type())
 	if err != nil {
 		return nil, err
 	}
 
-	for _, secDbMatch := range secDbMatches {
-		secDbCandidates[secDbMatch.Vulnerability.ID] = append(secDbCandidates[secDbMatch.Vulnerability.ID], secDbMatch)
+	cpeMatches, err := m.cpeMatchesWithoutSecDbFixes(store, d, p)
```

Comment:
I think I'm getting confused on the use of "CPE" vs. "NVD". My understanding of the "set operation" required for correct alpine matching was that it should specifically use NVD (in addition to Alpine's SecDB), as opposed to any data source that uses CPEs. With that assumption, I was expecting to see references to "NVD" here, as opposed to "CPE".

Having said that, I do also see references to NVD in this implementation. It would help my understanding to see consistency among these usages, so the answer to this question becomes clear: "Does Alpine matching _need_ NVD in particular? Or will any CPE-based data source do fine, and NVD is just one example of a data source that meets this requirement?"

Reply:
> Does Alpine matching need NVD in particular? Or will any CPE-based data source do fine?...

Any CPE-based data source will do fine 👍 

The intention of the `vulnerability.StoreAdapter` is to abstract away data source distinctions when asking for vulnerabilities (e.g. "get me vulnerabilities from SecDB") and instead frame it in a more agnostic way (e.g. "get me vulnerabilities for my distro"... where any and all sources will do) . The other matchers using the common CPE matcher are oriented the same way. In this way, when more data sources are added and indexed by CPEs, we don't need to make code adjustments.

Reply:
Okay, cool thanks! So that means the instances of "CPE" make sense. In that case, it's the remaining references like "find NVD matches" that were making me doubt the data source agnosticism.

Reply:
I think we can consider NVD or similar as "upstream" sources as opposed to vendor sources. That is the logical distinction for the matching behavior.